### PR TITLE
Pass HTTP proxy config during encryption

### PR DESCRIPTION
**Pull request by bburtin**
Add a new --proxy option for specifying one or more HTTPS proxies to use
during encryption.  Proxies are specified as host:port.  When proxies
are specified, we generate a proxy.yaml file and pass it to the
encryptor instance via user data.

To do:
* Specify HTTPS proxy when updating an AMI
* Read proxy.yaml from a file

### CODE REVIEW

File: brkt_cli/user_data.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import re
+from email import charset
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+import yaml
+
+# The directory for files saved on the Metavisor. We require that the dest
+# path for all --brkt-files be within this directory.
+BRKT_FILE_PREFIX = '/var/brkt/instance_config'
```
Comment (dramdass) [inline brkt_cli/user_data.py]: Do we use this? Also I think we're technically putting the yaml file in /var/brkt/ami_config/ here

**Comment by dramdass** [comment]
Code looks good! Nice job on the tests too. One minor comment

File: test.py [hunk 2]
```diff
@@ -1248,3 +1252,55 @@ def test_is_later_version_available(self):
...
+
+class TestProxy(unittest.TestCase):
+
+    def test_parse_proxies(self):
+        """ Test parsing host:port strings to Proxy objects.
+        """
+        # Valid
+        proxies = brkt_cli._parse_proxies(
+            'example1.com:8001',
+            'example2.com:8002'
```
Comment (bonzotronix) [inline test.py]: Might as well add an IP address or two here.  That'll be the most common case, FWIW.

File: brkt_cli/user_data.py (hunk 1 above)
Comment (bonzotronix) [inline brkt_cli/user_data.py]: Yep.  /var/brkt/ami_config/

File: brkt_cli/__init__.py [hunk 3]
```diff
@@ -285,6 +310,11 @@ def command_encrypt_ami(values, log):
     if values.brkt_env:
         brkt_env = _parse_brkt_env(values.brkt_env)
 
+    proxy_config = None
+    if values.proxies:
+        proxies = _parse_proxies(*values.proxies)
```
Comment (bonzotronix) [inline brkt_cli/__init__.py]: Maybe error out (here or elsewhere) if len(proxies) == 0.

**Comment by bonzotronix** [comment]
FTP.  Looks great overall.

File: brkt_cli/user_data.py (hunk 1 above)
Comment (bburtin) [inline brkt_cli/user_data.py]: Copy/paste cruft.  I'll remove it.

File: test.py (hunk 2 above)
Comment (bburtin) [inline test.py]: Done.

File: brkt_cli/__init__.py (hunk 3 above)
Comment (bburtin) [inline brkt_cli/__init__.py]: I don't think this would happen, due to the `if values.proxies` above.

**Pull request merged**